### DISCUSSION TITLE
"Chat with current note: Use your chatbot to reference and engage within your current note."

### DISCUSSION THREAD

**Discussion by gerroon** [longy2k/obsidian-bmo-chatbot#25] [Q&A]
Hi 
I have Ollama and this plugin working as a chat, but I do not know how you do what you listed as a feature which I put as the topic. Can you explain how I can chat with the note please?

**Comment by longy2k** [comment]
Oh, I should clarify that.

When the 'Reference Current Note' feature is on, it will allow the chatbot to read the current note that you have opened in the view (so, basically, the note that you are currently typing in Obsidian).

You can use the feature to ask the chatbot to look at your current active note and ask it anything like clarifying something you wrote awhile back, summarizing your note in bullet point, and etc.

You can enable this in the settings tab or type '/ref on' to activate it. Type '/ref off' to deactivate.

There is also a red or green indicator to show you if the activate note is currently being referenced. It only references markdown files.

**Comment by gerroon** [comment]
Thanks, but that does not seem to work or I do not know how to interact with it. I enabled the ref mode and asked "summarize the note" it kept summarizing stuff about my previous question.

Are there special commands or sentences that I need to use to interact with the open document?

Here is as screenshot, as you can see it is summarizing the question above, not the active note. And the active note has nothing to do with vector databases.


![image](https://github.com/longy2k/obsidian-bmo-chatbot/assets/8519469/704e9113-552b-4d91-a864-e2f720cdd197)

Also making sure the ref mode on (which was on already) but still the same issue

![image](https://github.com/longy2k/obsidian-bmo-chatbot/assets/8519469/172e209b-6f7b-4c8b-bb25-3b27ef83fe5d)

**Comment by twalderman** [comment]
Try linking the "current note" tab with the BMO tab.

**Reply by gerroon** [reply]
Thanks for the reply but how do I do it?

**Comment by twalderman** [comment]
right click on the BMO chat tab and you will see  "link with tab" then select the note tab you want to link with.

**Reply by gerroon** [reply]
Thanks, I did not know that but it did not work, as you can see it is linked. It still talks about vector databases

![image](https://github.com/longy2k/obsidian-bmo-chatbot/assets/8519469/ef2b40f0-8738-49ce-933b-e08552ed10f3)

**Comment by twalderman** [comment]
Have you tried this with a different model, specifically OpenAI or Anthropic?  I can confirm the functionality using Claude. To my knowledge it doesnt require a keyword or phrase to access the current document.  If you are able to see the text being sent to Ollama serve, you might be able to see if the current note is in the context window.

**Reply by gerroon** [reply]
I do not use commercial models. I am only testing it local models. I am able to send queries and get back results from Ollama in BMO  as you see in the screenshots.

The vector database answer in the screenshots was generated by Ollama sent from BMO (top screenshot), so we know it works.

**Comment by twalderman** [comment]
Unfortunately, I havent been able to get ollama to work with BMO on my mac.  perhaps there is something unique to that method that is not working.

**Reply by gerroon** [reply]
no problem, thanks for your help anyway.

**Comment by longy2k** [comment]
Local models are trickier to handle. 

When it comes to referencing the active note while having a previous conversation, I found it best to be specific in what it is that you want to summarize. 

For example, you asked "what is a vector database", then you want to summarize a note from a completely different topic, ask it to summarize that topic. If the topic is about cute kittens, ask something like "Summarize the note about cute kittens."

Also, I would recommend clearing your message history often, especially if you are switching context, using the commands `/clear` or `/c`. That way, the models that you use will be less confused.

Let me know if that helps!

**Comment by twalderman** [comment]
I got ollama configured correctly now.  I can confirm it is working with current note. I placed a phrase of unique nonsense in the middle of the current note and asked about it in the BMO chat.  Dolphin-Mistrel was able to respond.

If it helps, here is my modelfile:

FROM ./dolphin-2.2.1-mistral-7b.Q5_K_M.gguf
TEMPLATE """{{ if and .First .System }}<|im_start|>system
{{ .System }}<|im_end|>
{{ end }}<|im_start|>user
{{ .Prompt }}<|im_end|>
<|im_start|>assistant
"""
PARAMETER stop <|im_end|>
PARAMETER num_ctx 4096
PARAMETER temperature 0.7
PARAMETER top_p 0.9
PARAMETER repeat_penalty 1.1
PARAMETER top_k 100

**Reply by gerroon** [reply]
Thank you for testing it out. I think it works now. It was definitely related to clearing the chat window  and linking. Maybe these could be mentioned in the readme. Thanks.
